src-tauri/src/image_analyzer.rs

```diff
@@ -10,23 +10,39 @@ use crate::{
 };
 
 #[derive(Debug, Default, Clone, Copy)]
-pub struct Hp {
+pub struct Stat {
     pub max_w: u32,
-    pub hp: u32,
+    pub value: u32,
 }
 
-impl PartialEq for Hp {
+impl PartialEq for Stat {
     fn eq(&self, other: &Self) -> bool {
-        self.hp == other.hp
+        self.value == other.value
     }
 }
 
-impl PartialOrd for Hp {
+impl PartialOrd for Stat {
     fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        Some(self.hp.cmp(&other.hp))
+        Some(self.value.cmp(&other.value))
     }
 }
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Bar {
+    pub max_search_x:u32,
+    pub max_search_y:u32,
+    pub refs: [[u8; 3]; 4],
+}
```

Comment:
Here you could implement `Default` explicitly because the `max_search_x` and `max_search_y` is always the same:
```rust
impl Default for Bar {
    fn default() -> Self {
        Self {
            max_search_x: 310,
			max_search_y: 120,
            refs: [],
        }
    }
}
```

And then you can write a `new` method where you just pass the `refs`:
```rust
impl Bar {
	pub fn new(colors: [[u8; 3]; 4]) -> Self {
		Self {
			refs: colors,
			...Self::default(),
		}
	}
}
```

I would also suggest renaming `Bar` to `StatusBarConfig` or something like that, to make its purpose more clear.